File: CodeChef/Self Learning/Basic Programming Concepts/Motivation.cpp
// CodeChef - IMDB - Motivation
// https://www.codechef.com/LP1TO201/problems/IMDB

#include "bits/stdc++.h"
using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int T;
    cin >> T;
    while(T--){
        int N, X;
        cin >> N >> X;
        int highest_rating = INT_MIN;
        while(N--){
            int S, R;
            cin >> S >> R;
            if(S <= X){
                highest_rating = max(highest_rating, R);
            }
        }
        cout << highest_rating << endl;
    }

}

File: CodeChef/Self Learning/Basic Programming Concepts/Motivation.py
# CodeChef - IMDB - Motivation
# https://www.codechef.com/LP1TO201/problems/IMDB


import sys, math
input = sys.stdin.readline

T = int(input())

for _ in range(T):
    N, X = map(int, input().split())
    highest_rating = -math.inf

    for _ in range(N):
        S, R = map(int, input().split())
        if S <= X:
            highest_rating = max(highest_rating, R)
    
    print(highest_rating)
